docs/root/configuration/http/http_filters/grpc_stats_filter.rst

```diff
@@ -0,0 +1,25 @@
+.. _config_http_filters_grpc_stats:
+
+gRPC Statistics
+===============
+
+* gRPC :ref:`architecture overview <arch_overview_grpc>`
+* :ref:`v2 API reference <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpFilter.name>`
+* This filter should be configured with the name *envoy.filters.http.grpc_stats*.
+
+This is a filter which enables telemetry of gRPC calls. Additionally, the
+filter detects message boundaries in the streaming gRPC calls and emits the
+message counts for both the request and the response. The filter emits
+statistics in the *cluster.<route target cluster>.grpc.* namespace.
+
+More info: wire format in `gRPC over HTTP/2 <https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md>`_.
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 1, 1, 2
+
+  <grpc service>.<grpc method>.success, Counter, Total successful service/method calls
+  <grpc service>.<grpc method>.failure, Counter, Total failed service/method calls
+  <grpc service>.<grpc method>.total, Counter, Total service/method calls
+  <grpc service>.<grpc method>.request_count, Counter, Total request message count for service/method calls
```

Comment:
IMO `request_count` is a bit confusing in the sense that it's unclear if we are talking about single requests vs. messages. Should we actually split this out into per-API call stats and then per-API streamed message stats and make that more clear?

Reply:
Maybe just rename it to `request_message_count`? I am not sure how to model this per individual call, feel like two-dimensional data. The total message count across calls is still useful

Reply:
Note quite: You know a request/API has started because of the initial headers and can have stats on that, and then you could also have per-message stats?

Reply:
So the total/failure/success should capture call-level stats. The request counts are only emitted if the initial headers are parsed to extract gRPC service/method. The only possible improvement I can imagine is that we count successful and failed message counts separately, but that would require delaying the stats emissions to the very end (which may be quite long for streaming calls).

Reply:
If you have call-level stats then sure let's switch this to request_message_count and response_message_count?

Reply:
PTAL. Let me know if I need to change something here.